### PR TITLE
Coinstac storage proxy

### PR DESCRIPTION
**Pull request by dylancwood**
Contains four commits:
'create instance of xhrAgent in client.js, not codegen-cleint' is not related to coinstac storage proxy.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -95,7 +102,16 @@
     "postinstall": "npm run build",
     "update-codegen": "node scripts/update-codegen.js",
     "generate-client": "node scripts/generate-client.js",
-    "postgenerate-client": "node node_modules/webpack/bin/webpack.js"
+    "postgenerate-client": "node node_modules/webpack/bin/webpack.js",
+    "lintfix": "jscs --fix src/**/*.js test/**/*.js",
+    "postlint": "jshint src/**/*.js test/**/*.js",
```
Comment (cdaringe) [inline package.json:103]: per your findings, this globbing doesn't always work.  :(  i think a `shopt` glob option must be set?  anyway, a week or so ago i published a version of coins-validate that drops the `/**/*.js` which our deps still successfully recurse!  :game_die: :game_die: :game_die: :game_die: :game_die:

File: src/lib/utils/get-plugins.js [hunk 2]
```diff
@@ -1,11 +1,10 @@
-'use strict';
+    'use strict';
```
Comment (cdaringe) [inline src/lib/utils/get-plugins.js]: > > !

File: src/lib/utils/get-plugins.js [hunk 3]
```diff
@@ -81,7 +79,11 @@ var plugins = [
         register: require('./response-formatter.js'),
         options: {
             excludeVarieties: ['view', 'file'],
-            excludePlugins: ['hapi-swagger', 'client-source']
+            excludePlugins: [
+                'hapi-swagger',
+                'client-source',
+                'coinstac-storage-proxy'
+            ]
```
Comment (cdaringe) [inline src/lib/utils/get-plugins.js:86]: _cool_ :+1: .  that's awesome that you can specify these

File: src/lib/utils/get-plugins.js (hunk 2 above)
Comment (dylancwood) [inline src/lib/utils/get-plugins.js]: hmm. that's odd. I'll fix it.

**Pull request merged**